### PR TITLE
Oddly written (including unreachable code) function in aws/routes/routeAwsSubaccountPath.go

### PR DESCRIPTION
**Pull request by GabrielRavier**
There's some unreachable code in patchAwsSubAccountRequestBody which just answers with an 200 OK response code and nothing else (whereas it's supposed to send back the AWS account information). I've
refactored the code around it too as the if-else usage in general was somewhat dubious too.

### CODE REVIEW

**Review by cruffinoni — APPROVED** [review]
LGTM but we will wait for @simonmeyerrr to review this one before merging since it's a different coding style.

**Review by simonmeyerrr — APPROVED** [review]

**Pull request merged**
